token/services/network/fabric/network.go

```diff
@@ -310,3 +317,63 @@ func (n *Network) SubscribeTxStatusChanges(txID string, listener driver.TxStatus
 func (n *Network) UnsubscribeTxStatusChanges(txID string, listener driver.TxStatusChangeListener) error {
 	return n.ch.Committer().UnsubscribeTxStatusChanges(txID, listener)
 }
+
+func (n *Network) ScanForKey(namespace string, startingTxID string, key string, timeout time.Duration, opts ...token.ServiceOption) ([]byte, error) {
+	var keyValue []byte
+	c, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	vault := n.ch.Vault()
+	if err := n.ch.Delivery().Scan(c, startingTxID, func(tx *fabric.ProcessedTransaction) (bool, error) {
+		logger.Debugf("scanning [%s]...", tx.TxID())
+
+		rws, err := vault.GetEphemeralRWSet(tx.Results())
+		if err != nil {
+			return false, err
+		}
+
+		found := false
+		for _, ns := range rws.Namespaces() {
+			if ns == namespace {
+				found = true
+				break
+			}
+		}
+		if !found {
+			logger.Debugf("scanning [%s] does not contain namespace [%s]", tx.TxID(), namespace)
+			return false, nil
+		}
+
+		ns := namespace
+		w := translator.New(tx.TxID(), NewRWSWrapper(rws), namespace)
+		for i := 0; i < rws.NumWrites(ns); i++ {
+			k, v, err := rws.GetWriteAt(ns, i)
+			if err != nil {
+				return false, err
+			}
+			subKey, err := w.GetTransferMetadataKeyWithSubKey(k)
+			if err != nil {
+				continue
+			}
+			if subKey == key {
+				keyValue = v
+				return true, nil
+			}
+		}
+		logger.Debugf("scanning for preimage on [%s] not found", tx.TxID())
```

Comment:
this shouldn't say preimage....should be more general

Reply:
indeed.